tempto-core/src/main/java/com/teradata/tempto/internal/initialization/RequirementsExpanderInterceptor.java

```diff
@@ -80,6 +88,49 @@ public static int getMethodsCountFromContext(ITestContext context)
         return allExpandedMethods;
     }
 
+    private void incrementSeenMethodsCount(List<IMethodInstance> newExpandedMethods)
+    {
+        for (IMethodInstance newExpandedMethod : newExpandedMethods) {
+            Optional<Object[][]> parametersForMethod = getParametersForMethod(newExpandedMethod);
+            if (parametersForMethod.isPresent()) {
+                seenMethodsCount += parametersForMethod.get().length;
+            } else {
+                seenMethodsCount++;
+            }
+        }
+    }
+
+    private Optional<Object[][]> getParametersForMethod(IMethodInstance method)
```

Comment:
Maybe put code responsible for data providers (this reflection code) into separate class and add proper testing.